app/src/main/java/com/github/libretube/ui/dialogs/UpdateAvailableDialog.kt

```diff
@@ -6,12 +6,21 @@ import android.os.Bundle
 import androidx.core.net.toUri
 import androidx.fragment.app.DialogFragment
 import com.github.libretube.R
+import com.github.libretube.constants.IntentData
 import com.github.libretube.obj.update.UpdateInfo
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import kotlinx.serialization.json.Json
 
-class UpdateAvailableDialog(
-    private val updateInfo: UpdateInfo
-) : DialogFragment() {
+class UpdateAvailableDialog : DialogFragment() {
+    private lateinit var updateInfo: UpdateInfo
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            val encodedString = it.getString(IntentData.updateInfo)!!
+            updateInfo = Json.decodeFromString(encodedString)
+        }
```

Comment:
It would be better to update the `UpdateInfo` class to implement the `Parcelable` interface instead (a custom `Parceler` implementation would be needed for `Instant`: https://developer.android.com/kotlin/parcelize#custom_parcelers). That way, this block could be replaced with the following:

```suggestion
        updateInfo = requireArguments().parcelable(IntentData.updateInfo)!!
```